web_src/js/components/RepoBranchTagDropdown.js

```diff
@@ -64,6 +75,26 @@ export function initRepoBranchTagDropdown(selector) {
       beforeMount() {
         this.noResults = this.$el.getAttribute('data-no-results');
         this.canCreateBranch = this.$el.getAttribute('data-can-create-branch') === 'true';
+        this.branchForm = this.$el.getAttribute('data-branch-form');
+        switch (this.$el.getAttribute('data-view-type')) {
+          case 'tree':
+            this.isViewTree = true;
+            break;
+          case 'tag':
+            this.isViewTag = true;
+            break;
+          default:
+            this.isViewBranch = true;
+            break;
+        }
+        this.refName = this.$el.getAttribute('data-ref-name');
+        this.branchURLPrefix = this.$el.getAttribute('data-branch-url-prefix');
+        this.branchURLSuffix = this.$el.getAttribute('data-branch-url-suffix');
+        this.tagURLPrefix = this.$el.getAttribute('data-tag-url-prefix');
+        this.tagURLSuffix = this.$el.getAttribute('data-tag-url-suffix');
+        this.setAction = this.$el.getAttribute('data-set-action') === 'true';
+        this.submitForm = this.$el.getAttribute('data-submit-form') === 'true';
```

Comment:
If we use `PageData` to pass these variables, these code can be simplified

Reply:
You are more than welcome to write the PR that makes that change.